core/src/main/java/org/elasticsearch/index/query/FuzzyQueryBuilder.java

```diff
@@ -258,63 +258,60 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
 
     public static Optional<FuzzyQueryBuilder> fromXContent(QueryParseContext parseContext) throws IOException {
         XContentParser parser = parseContext.parser();
-
```

Comment:
this type of change is mechanical and the same for some of the parsers that didn't loop till the END_OBJECT. That would cause the query to not be completely read in case multiple fields are presented. That would be caught by the new check I am introducing in QueryParseContext#parseInnerQueryBuilder, but I think it is better to change the way we parse those queries to exhaust their own object at all times.